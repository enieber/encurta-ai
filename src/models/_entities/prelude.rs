//! `SeaORM` Entity, @generated by sea-orm-codegen 1.1.7

pub use super::routers::Entity as Routers;
pub use super::users::Entity as Users;
