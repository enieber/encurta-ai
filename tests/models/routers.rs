use loco_rs::testing::prelude::*;
use encurta_ai::{
    app::App,
    models::routers::{cleanup_routers_model, Model, RoutersParams},
};
use insta::assert_debug_snapshot;
use insta::with_settings;
use serial_test::serial;

macro_rules! configure_insta {
    ($($expr:expr),*) => {
        let mut settings = insta::Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("routers");
        let _guard = settings.bind_to_scope();
    };
}


#[tokio::test]
#[serial]
async fn can_save_link() {
    configure_insta!();

    let boot = boot_test::<App>()
        .await
        .expect("Failed to boot test application");

    let params = RoutersParams {
        link: "https://mapeamento.softwarelivre.tec.br/".to_string(),
    };

    let res = Model::save_link(&boot.app_context.db, &params).await;

    
    with_settings!({
        filters => cleanup_routers_model()
    }, {
        assert_debug_snapshot!(res);
    });
}
