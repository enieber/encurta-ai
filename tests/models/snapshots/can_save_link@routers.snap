---
source: tests/models/routers.rs
expression: res
---
Ok(
    Model {
        created_at: DATE,
        updated_at: DATE,
        id: ID
        link: "https://mapeamento.softwarelivre.tec.br/",
        hash: "6746943",
    },
)
